sparkle/src/components/Searchbar.tsx

```diff
@@ -17,78 +17,89 @@ const iconClasses = {
   md: "s-pr-4",
 };
 
-export function Searchbar({
-  placeholder,
-  value,
-  onChange,
-  onKeyDown,
-  name,
-  size = "sm",
-  disabled = false,
-  className = "",
-}: {
-  placeholder: string;
-  value: string | null;
-  onChange?: (value: string) => void;
-  onKeyDown?: (e: React.KeyboardEvent<HTMLInputElement>) => void;
-  name: string;
-  size?: "xs" | "sm" | "md";
-  disabled?: boolean;
-  className?: string;
-}) {
-  const clearInputField = () => {
-    onChange?.("");
-  };
+export const Searchbar = forwardRef<
+  HTMLInputElement,
```

Comment:
Why this type? this is a div at the top? Is it needed?

Reply:
Why do we need the forwardRef machinery? How is the focus event passed down to the input?

Reply:
I tried without this type and add a bunch of errors telling me to use this type. But curious to hear if there's a better to implement focus 🤓

Reply:
@flvndvd any suggestion?

Reply:
I guess it is typed as the element we're pointing the ref to which makes sense here. Didn't get that we were forwarding the ref duh :p

Reply:
Yep this is the right way to do it 🙃.

Reply:
Actually, most of our design system should use `forwardRef` so we can have more control from the call sites.